mne_bids/mne_bids.py

```diff
@@ -224,10 +244,21 @@ def _participants_tsv(raw, subject_id, group, fname, verbose):
         Name of group participant belongs to.
     fname : str
         Filename to save the participants.tsv to.
+    write_mode : str, one of ('append', 'overwrite', 'error')
+        How the file should be handled if is exists already.
+        Defaults to 'append'.
+        If write_mode == 'append' the previously existing file will have the
+        new data added to it, overwriting any duplicate entries found.
+        If write_mode == 'overwrite' the previously existing file will be
+        removed and replaced with the new data.
+        If write_mode == 'error' an `OSError` will be raised.
     verbose : bool
         Set verbose output to true or false.
 
     """
+    if write_mode == 'error':
+        _check_file_exists(fname)
```

Comment:
why don't you put the condition above also in the private function?

Reply:
I originally wrote it so that the write mode was also in the function, but I was struggling to rationalise the name and functionality. As it stands I am still not particularly happy with it, but by including the write mode as well I feel it became too far removed from a concise name.
It would look something like:
```python
def _check_file_exists(fname, write_mode):
    if op.exists(fname) and write_mode=='error':
        raise OSError(errno.EEXIST, '"%s" already exists. Please set '
                      'write_mode to "overwrite" or "append".' % fname)
    return False
```
Someone reading the code would expect that if `write_mode != 'error'` then it should do something else. So would I get it to return true if it does exist but `write_mode != 'error'`?
Other names I thought of were things like `_error_on_exist` (for example, but that also wouldn't require the `write_mode`)

Reply:
okay fair enough